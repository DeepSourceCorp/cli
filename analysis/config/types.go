package config

import "github.com/deepsourcelabs/cli/analysis/lsp"

///////////////////////////////////////////////////////
// DSConfig is the struct for .deepsource.toml file //
/////////////////////////////////////////////////////

type DSConfig struct {
	Version         int           `mapstructure:"version,omitempty" json:"version" toml:"version"`
	ExcludePatterns []string      `mapstructure:"exclude_patterns,omitempty" json:"exclude_patterns,omitempty" toml:"exclude_patterns"`
	TestPatterns    []string      `mapstructure:"test_patterns,omitempty" json:"test_patterns,omitempty" toml:"test_patterns"`
	Analyzers       []Analyzer    `mapstructure:"analyzers,omitempty" json:"analyzers,omitempty" toml:"analyzers"`
	Transformers    []Transformer `mapstructure:"transformers,omitempty" json:"transformers,omitempty" toml:"transformers"`
}

type Analyzer struct {
	Name                string      `mapstructure:"name,omitempty" json:"name,omitempty" toml:"name"`
	RuntimeVersion      string      `mapstructure:"runtime_version,omitempty" json:"runtime_version,omitempty" toml:"runtime_version,omitempty"`
	Enabled             bool        `mapstructure:"enabled,omitempty" json:"enabled" toml:"enabled"`
	DependencyFilePaths []string    `mapstructure:"dependency_file_paths,omitempty" json:"dependency_file_paths,omitempty"`
	Meta                interface{} `mapstructure:"meta,omitempty" json:"meta,omitempty" toml:"meta"`
	Thresholds          interface{} `mapstructure:"thresholds,omitempty" json:"thresholds,omitempty"`
}

type Transformer struct {
	Name    string `mapstructure:"name,omitempty" json:"name,omitempty" toml:"name"`
	Enabled bool   `mapstructure:"enabled,omitempty" json:"enabled,omitempty" toml:"enabled"`
}

///////////////////////////////////////
// LSP based Analysis Config Types  //
/////////////////////////////////////

type AnalysisConfig struct {
	Files           []lsp.TextDocumentItem `json:"files"`
	TestFiles       []lsp.TextDocumentItem `json:"test_files"`
	ExcludedFiles   []lsp.TextDocumentItem `json:"excluded_files"`
	ExcludePatterns []string               `json:"exclude_patterns"`
	TestPatterns    []string               `json:"test_patterns"`
	AnalyzerMeta    interface{}            `json:"analyzer_meta"`
}

//////////////////////////////////////
// LSP based Analysis Result Types  //
/////////////////////////////////////

type Namespace struct {
	Key   string  `json:"key"`
	Value float64 `json:"value"`
}

type Metric struct {
	MetricCode string      `json:"metric_code"`
	Namespaces []Namespace `json:"namespaces"`
}

type AnalysisResult struct {
	Issues   []lsp.Diagnostic `json:"issues"`
	Metrics  []Metric         `json:"metrics,omitempty"`
	IsPassed bool             `json:"is_passed"`
	Errors   []Error          `json:"errors"`
	// Errors    []lsp.Diagnostic `json:"errors"`
	ExtraData interface{} `json:"extra_data"`
}

////////////////////////////////////
// Default Analysis Result Types  //
///////////////////////////////////

type Error struct {
	HMessage string `json:"hmessage"`
	Level    int    `json:"level"`
}

type Coordinate struct {
	Line   int `json:"line"`
	Column int `json:"column"`
}

type Position struct {
	Begin Coordinate `json:"begin"`
	End   Coordinate `json:"end"`
}

type Location struct {
	Path     string   `json:"path"`
	Position Position `json:"position"`
}

type Issue struct {
	Code     string   `json:"issue_code"`
	Title    string   `json:"issue_text"`
	Location Location `json:"location"`
}

type DefaultAnalysisResult struct {
	Issues    []Issue     `json:"issues"`
	Metrics   []Metric    `json:"metrics,omitempty"`
	IsPassed  bool        `json:"is_passed"`
	Errors    []Error     `json:"errors"`
	ExtraData interface{} `json:"extra_data"`
}
