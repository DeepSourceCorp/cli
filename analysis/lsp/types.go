package lsp

//////////////////////////////
// Document identity types  //
/////////////////////////////

type DocumentURI string

type TextDocumentItem struct {
	URI        DocumentURI `json:"uri"`
	LanguageID string      `json:"languageID,omitempty"`
	Version    int         `json:"version,omitempty"`
	Text       string      `json:"text,omitempty"`
}

///////////////////////////////
// Diagnostic related types //
/////////////////////////////

type DiagnosticSeverity int

const (
	Error       DiagnosticSeverity = 1
	Warning     DiagnosticSeverity = 2
	Information DiagnosticSeverity = 3
	Hint        DiagnosticSeverity = 4
)

type Position struct {
	Line      int `json:"line"`
	Character int `json:"character"`
}
type Range struct {
	Start Position `json:"start"`
	End   Position `json:"end"`
}

type Location struct {
	URI   string `json:"uri"`
	Range Range  `json:"range"`
}

type DiagnosticRelatedInformation struct {
	Location Location `json:"location"`
	Message  string   `json:"message"`
}

type Diagnostic struct {
	Range    Range              `json:"range"`
	Severity DiagnosticSeverity `json:"severity,omitempty"`
	Code     string             `json:"code,omitempty"`
	Source   string             `json:"source,omitempty"`
	Message  string             `json:"message"`

	/**
	 * An array of related diagnostic information, e.g. when symbol-names within
	 * a scope collide all definitions can be marked via this property.
	 * var a,b
	 * a := 2
	 * Issues in line 1 and 2 are related.
	 */
	RelatedInformation []DiagnosticRelatedInformation `json:"relatedInformation"`
}
