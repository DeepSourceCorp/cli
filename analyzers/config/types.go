package config

type Analysis struct {
	Command string `toml:"command" json:"command" validate:"required"`
}

// TODO: Remove the comment when we start support for Autofix.
// type Autofix struct {
//     Command string `toml:"command" json:"command" analyzertoml:"required"`
// }

type Build struct {
	Builder    string `toml:"builder" json:"builder" validate:"omitempty"`
	Dockerfile string `toml:"dockerfile" json:"dockerfile,omitempty" validate:"omitempty"`
	Script     string `toml:"script" json:"script" validate:"omitempty"`
}

type URL struct {
	Source        string `toml:"source" json:"source" validate:"omitempty,url"`
	Documentation string `toml:"documentation" json:"documentation,omitempty" validate:"omitempty,url"`
	BugTracker    string `toml:"bug_tracker" json:"bug_tracker" validate:"omitempty,url"`
}

type Test struct {
	Command string `toml:"command" json:"command" validate:"omitempty"`
}

// analyzer.toml type
type AnalyzerMetadata struct {
	// Analyzer specific data
	Name                 string            `toml:"name" json:"name" validate:"required"`
	Shortcode            string            `toml:"shortcode" json:"shortcode" validate:"required"` // New
	Description          string            `toml:"description" json:"description" validate:"required"`
	Category             string            `toml:"category" json:"category" validate:"required"`
	Tags                 []string          `toml:"tags" json:"tags,omitempty" validate:"omitempty"`
	URL                  URL               `toml:"urls" json:"urls,omitempty" validate:"omitempty"`
	EnvironmentVariables map[string]string `toml:"environment_variables" json:"environment_variables,omitempty" validate:"omitempty"`

	// Analyzer, Autofix and Transformer config
	Analysis Analysis `toml:"analysis" json:"analysis" validate:"required"`
	// Autofix Autofix `toml:"autofix" json:"autofix" validate:"omitempty"`

	// Build steps needed to build the analyzer
	Build Build `toml:"build" json:"build" validate:"omitempty"`

	// Test command for running the testing workflow
	Test Test `toml:"test" json:"test" validate:"omitempty"`
}

// Analyzer issue description
type AnalyzerIssue struct {
	Shortcode   string `validate:"omitempty"`
	Title       string `toml:"title" json:"title" validate:"required"`
	Description string `toml:"description" json:"description" validate:"required"`
	Category    string `toml:"category" json:"category" validate:"required"`
}
