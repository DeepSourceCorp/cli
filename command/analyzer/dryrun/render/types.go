package render

import (
	"html/template"
	"time"

	"github.com/deepsourcelabs/cli/types"
	"github.com/labstack/echo/v4"
)

type RunSummary struct {
	RunDuration       string // Time taken to complete analysis.
	TimeSinceRun      string // Time elapsed since the completion of the analysis run.
	AnalysisStartTime time.Time
	AnalysisEndTime   time.Time
}

type VCSInfo struct {
	Branch      string // VCS branch of the Analyzer.
	CommitSHA   string // The latest commit SHA of the Analyzer.
	VersionDiff string // The string specifying the status of Analyzer w.r.t previous version.
}

type OccurenceData struct {
	IssueMeta  types.AnalyzerIssue // Contains the data stored in issue TOMLs for the respective issue.
	Files      []string            // Files where this issue has been reported.
	FilesInfo  string              // The string containing the data of which files the issue has been reported in.
	Occurences []types.Issue       // The slice of occurences for a certain issue code.
}

type ResultData struct {
	UniqueIssuesCount     int                      // The unique issues count.
	TotalOccurences       int                      // Total issues reported by the Analyzer.
	SourcePath            string                   // The path where the source code to be analyzer is stored.
	IssuesOccurenceMap    map[string]OccurenceData // The map of issue code to its occurences data.
	IssueCategoryCountMap map[string]int           // The map of issue category to the count of the issues of that category.
	AnalysisResult        types.AnalysisResult     // The analysis result post running processors.
	MetricsMap            map[string]float64       // The map of metric names to their values.
	RenderedSourceCode    []template.HTML          // The slice containing the source code snippets for each occurence.
}

type ResultRenderOpts struct {
	EchoServer           *echo.Echo         // The Echo server instance to run the renderer server.
	Template             *template.Template // The go template field so that it can be accessible in `route.go` as well.
	PageTitle            string             // The title of the HTML page.
	AnalyzerShortcode    string             // The shortcode of the Analyzer.
	VCSInfo              VCSInfo            // The VCS information of the Analyzer.
	Summary              RunSummary         // The run summary.
	AnalysisResultData   ResultData         // The analysis result data.
	SelectedIssueCode    string             // The field used to recognize which issue code the user has clicked on to check its occurences.
	SelectedCategory     string             // The field used to recognize which category the user has clicked to filter the issues based on it.
	IssueCategoryNameMap map[string]string  // The map used to route category names to their codes. Eg: `Documentation`->`doc`.
	MetricNameMap        map[string]string  // The map of metrics shortcodes with their names.
}
