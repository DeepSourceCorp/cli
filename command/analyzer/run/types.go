package run

type AnalysisConfig struct {
	Files           []string    `json:"files"`
	ExcludePatterns []string    `json:"exclude_patterns"`
	ExcludeFiles    []string    `json:"exclude_files"`
	TestFiles       []string    `json:"test_files"`
	TestPatterns    []string    `json:"test_patterns"`
	AnalyzerMeta    interface{} `json:"analyzer_meta"`
}
