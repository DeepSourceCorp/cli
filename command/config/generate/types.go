package generate

// DSConfig is the struct for .deepsource.toml file
type Analyzer struct {
	Name                string      `toml:"name,omitempty" json:"name,omitempty"`
	RuntimeVersion      string      `toml:"runtime_version,omitempty" json:"runtime_version,omitempty"`
	Enabled             bool        `toml:"enabled" json:"enabled"`
	DependencyFilePaths []string    `toml:"dependency_file_paths,omitempty" json:"dependency_file_paths,omitempty"`
	Meta                interface{} `toml:"meta,omitempty" json:"meta,omitempty"`
	Thresholds          interface{} `toml:"thresholds,omitempty" json:"thresholds,omitempty"`
}

type Transformer struct {
	Name    string `toml:"name" json:"name"`
	Enabled bool   `toml:"enabled" json:"enabled"`
}

type DSConfig struct {
	Version         int           `toml:"version" json:"version"`
	ExcludePatterns []string      `toml:"exclude_patterns" json:"exclude_patterns,omitempty"`
	TestPatterns    []string      `toml:"test_patterns" json:"test_patterns,omitempty"`
	Analyzers       []Analyzer    `toml:"analyzers,omitempty" json:"analyzers,omitempty"`
	Transformers    []Transformer `toml:"transformers,omitempty" json:"transformers,omitempty"`
}
