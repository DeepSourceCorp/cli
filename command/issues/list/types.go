package list

// custom types for JSON marshaling

type IssueJSON struct {
	Analyzer       string       `json:"analyzer"`
	IssueCode      string       `json:"issue_code"`
	IssueTitle     string       `json:"issue_title"`
	OccurenceTitle string       `json:"occurence_title"`
	IssueCategory  string       `json:"issue_category"`
	Location       LocationJSON `json:"location"`
}

type LocationJSON struct {
	Path     string       `json:"path"`     // The filepath where the issue is reported
	Position PositionJSON `json:"position"` // The position info where the issue is raised
}

type PositionJSON struct {
	Begin LineColumn `json:"begin"` // The line where the code covered under the issue starts
	End   LineColumn `json:"end"`   // The line where the code covered under the issue starts
}

type LineColumn struct {
	Line   int `json:"line"`
	Column int `json:"column"`
}
