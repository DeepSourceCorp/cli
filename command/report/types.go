package report

// ReportQueryInput is the schema for variables of artifacts
// report GraphQL query
type ReportQueryInput struct {
	AccessToken       string      `json:"accessToken"`
	CommitOID         string      `json:"commitOid"`
	ReporterName      string      `json:"reporter"`
	ReporterVersion   string      `json:"reporterVersion"`
	Key               string      `json:"key"`
	Data              string      `json:"data"`
	AnalyzerShortcode string      `json:"analyzer"`
	Metadata          interface{} `json:"metadata,omitempty"`
}

// ReportQueryInput is the structure of artifacts report
// GraphQL query
type ReportQuery struct {
	Query     string `json:"query"`
	Variables struct {
		Input ReportQueryInput `json:"input"`
	} `json:"variables"`
}

// QueryResponse is the response returned by artifacts report
// GraphQL query
type QueryResponse struct {
	Data struct {
		CreateArtifact struct {
			Error string `json:"error"`
			Ok    bool   `json:"ok"`
		} `json:"createArtifact"`
	} `json:"data"`
}
