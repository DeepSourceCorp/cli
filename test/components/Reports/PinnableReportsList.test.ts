import { ZMenu } from '@deepsource/zeal'
import { render } from '@testing-library/vue'
import { createLocalVue, shallowMount } from '@vue/test-utils'
import VTooltip from 'floating-vue'

import PinnableReportsList from '~/components/Reports/PinnableReportsList.vue'
import { IPinnableReportItem, ReportMetadataFilterT, ReportPageT } from '~/types/reportTypes'
import { ReportLevel } from '~/types/types'

interface IPinnableReportsList extends Vue {
  // Data properties
  collapsibleHeaderAnimations: Record<ReportPageT, string>

  // Computed properties
  complianceItems: Array<IPinnableReportItem>
  cookieIdentifier: string
  insightsItems: Array<IPinnableReportItem>

  // Methods
  closeMenu: () => void
  isCurrentlyPinnedReport: (reportItem: IPinnableReportItem) => boolean
  expandCurrentlySelectedCollapsible: () => void
  menuToggleHandler(isOpen: boolean): void
  updateCollapsibleMetadata(reportItem: IPinnableReportItem, isOpen: boolean): void
  updatePinnedReport: (reportItem: IPinnableReportItem) => void
}

describe('[[ PinnableReportsList ]]', () => {
  const baseProps = {
    currentSelection: 'issue-distribution-analyzer',
    level: ReportLevel.Owner,
    owner: 'deepsourcelabs',
    pinnedReports: [
      {
        key: 'sans-top-25',
        metadata: null
      },
      {
        key: 'issue-distribution',
        metadata: {
          filter: 'issue-distribution-analyzer'
        }
      },
      {
        key: 'code-coverage',
        metadata: null
      },
      {
        key: 'owasp-top-10',
        metadata: null
      }
    ],
    provider: 'gh',
    reportSlot: 1
  }

  const mocks = {
    $cookies: {
      remove: jest.fn()
    },
    $emit: jest.fn()
  }

  const stubs = {
    ZButton: true,
    ZIcon: true,
    ZMenu, // Use original `ZMenu` component being the container
    ZMenuSection: true,
    ZTag: true
  }

  // Returns a Wrapper that contains the rendered Vue component
  const getInstance = (props = {}) => {
    const localVue = createLocalVue()
    localVue.use(VTooltip)

    const { vm } = shallowMount(PinnableReportsList, {
      mocks,
      propsData: { ...baseProps, ...props },
      stubs,
      localVue
    })

    return vm as IPinnableReportsList
  }

  test('renders the list of pinnable reports as menu items', () => {
    const { html } = render(
      PinnableReportsList,
      {
        mocks,
        propsData: { ...baseProps },
        stubs
      },
      (vue) => {
        vue.use(VTooltip)
      }
    )

    expect(html()).toMatchSnapshot()
  })

  test('`cookieIdentifier` returns the appropriate format for distribution-based reports', () => {
    const instance = getInstance()

    const { owner, pinnedReports, provider, reportSlot } = baseProps

    // Grab the current report item
    const reportItem = pinnedReports[reportSlot]

    // Cookie identifier has the convention `provider_owner_reportType_date-range-filter_report-slot`
    // Distribution-based reports have `metadata` with the `filter` type
    expect(instance.cookieIdentifier).toBe(
      `${provider}_${owner}_${
        reportItem.metadata?.filter as ReportMetadataFilterT
      }_date-range-filter_${reportSlot}`
    )
  })

  test('`cookieIdentifier` returns the appropriate format for compliance-based reports', () => {
    // Specify the current selection as `owasp-top-10` report
    // It comes at slot `3` as part of `baseProps`
    const reportSlot = 3
    const instance = getInstance({ currentSelection: 'owasp-top-10', reportSlot })

    const { owner, pinnedReports, provider } = baseProps

    // Grab the current report item
    const reportItem = pinnedReports[reportSlot]

    // Cookie identifier has the convention `provider_owner_reportType_date-range-filter_report-slot`
    // Compliance-based reports has `metadata` specified as `null`
    expect(instance.cookieIdentifier).toBe(
      `${provider}_${owner}_${reportItem.key}_date-range-filter_${reportSlot}`
    )
  })

  test('`cookieIdentifier` returns the appropriate format for code-coverage report', () => {
    // `code-coverage` report comes at slot `2` as part of `baseProps`
    const reportSlot = 2
    const instance = getInstance({ currentSelection: 'code-coverage', reportSlot })

    const { owner, provider } = baseProps

    // For `code-coverage` report, it is `provider_owner_pinned-code-coverage_sort-type_report-slot`
    expect(instance.cookieIdentifier).toBe(
      `${provider}_${owner}_pinned-code-coverage-sort-type_${reportSlot}`
    )
  })

  test('`isCurrentlyPinnedReport` method returns the status regarding the current match on supplying a distribution-based report', () => {
    const instance = getInstance()

    const reportKey = ReportPageT.DISTRIBUTION

    // Find the entry corresponding to `issue-distribution` from `insightsItem`
    const reportItem = instance.insightsItems.find(
      ({ key }) => key === reportKey
    ) as IPinnableReportItem

    const pinnedStatus = instance.isCurrentlyPinnedReport(reportItem)
    expect(pinnedStatus).toBe(true)
  })

  test('`isCurrentlyPinnedReport` method returns the status regarding the current match on supplying a compliance-based report', () => {
    // Specify the current selection as `owasp-top-10` report
    // It comes at slot `3` as part of `baseProps`
    const instance = getInstance({ currentSelection: 'owasp-top-10', reportSlot: 3 })

    const reportKey = ReportPageT.OWASP_TOP_10

    // Find the entry corresponding to `owasp-top-10` from `complianceItem`
    const reportItem = instance.complianceItems.find(
      ({ key }) => key === reportKey
    ) as IPinnableReportItem

    const pinnedStatus = instance.isCurrentlyPinnedReport(reportItem)
    expect(pinnedStatus).toBe(true)
  })

  test('`menuToggleHandler` method returns early if the menu is getting closed', () => {
    const instance = getInstance()

    // Set spy
    instance.expandCurrentlySelectedCollapsible = jest.fn()

    // Invoke the `menuToggleHandler` method supplying `isOpen` as `false`
    instance.menuToggleHandler(false)

    // Assertion
    expect(instance.expandCurrentlySelectedCollapsible).not.toBeCalled()
  })

  test('`menuToggleHandler` method resets chevron animations and expands the currently selected collapsible', () => {
    const instance = getInstance()

    // Set spy
    instance.expandCurrentlySelectedCollapsible = jest.fn()

    // Initial value for `collapsibleHeaderAnimations`
    const initialCollapsibleHeaderAnimations = {
      [ReportPageT.DISTRIBUTION]: '',
      [ReportPageT.ISSUES_PREVENTED]: ''
    }

    // Invoke the `menuToggleHandler` method supplying `isOpen` as `true`
    instance.menuToggleHandler(true)

    // Assertions
    expect(instance.collapsibleHeaderAnimations).toEqual(initialCollapsibleHeaderAnimations)
    expect(instance.expandCurrentlySelectedCollapsible).toBeCalled()
  })

  test('`updatePinnedReport` method returns early if the report item supplied matches current selection', () => {
    // `currentSelection` defaults to `issue-distribution-analyzer`
    const instance = getInstance()

    // Set spy
    instance.closeMenu = jest.fn()

    const reportKey = ReportPageT.DISTRIBUTION

    // Find the entry corresponding to `issue-distribution` from `insightsItem`
    const reportItem = instance.insightsItems.find(
      ({ key }) => key === reportKey
    ) as IPinnableReportItem

    // Invoke the `updatePinnedReport` method
    instance.updatePinnedReport({
      ...reportItem,
      metadata: reportItem.metadataItems?.[0] // `By Analyzer`
    } as IPinnableReportItem)

    // Assertions
    expect(instance.closeMenu).toBeCalled()
    expect(mocks.$cookies.remove).not.toBeCalled()
    expect(mocks.$emit).not.toBeCalled()
  })

  test('`updatePinnedReport` emits an event by the name `update-pinned-reports` supplying the updated `pinnedReports` array', () => {
    // `currentSelection` defaults to `issue-distribution-analyzer`
    const instance = getInstance()

    // Set spy
    instance.closeMenu = jest.fn()

    const newReportKey = ReportPageT.DISTRIBUTION
    const oldReportKey = ReportPageT.OWASP_TOP_10

    // Find the entry corresponding to `issue-distribution` from `insightsItem`
    const oldReportItem = instance.insightsItems.find(
      ({ key }) => key === newReportKey
    ) as IPinnableReportItem

    // Find the entry corresponding to `owasp-top-10` from `complianceItem`
    const newReportItem = instance.complianceItems.find(
      ({ key }) => key === oldReportKey
    ) as IPinnableReportItem

    // Invoke the `updatePinnedReport` method
    instance.updatePinnedReport(newReportItem)

    // Assert for the `closeMenu` method invocation
    expect(instance.closeMenu).toBeCalled()

    const { owner, provider, reportSlot } = baseProps

    expect(mocks.$cookies.remove).toBeCalledWith(
      `${provider}_${owner}_${
        oldReportItem.metadataItems?.[0]?.filter as ReportMetadataFilterT
      }_date-range-filter_${reportSlot}` // `By Analyzer`
    )

    const { key, metadata } = newReportItem

    const newPinnedReports = [...baseProps.pinnedReports]
    newPinnedReports[reportSlot] = { key, metadata }
    expect(mocks.$emit).toBeCalledWith(
      'update-pinned-reports',
      newPinnedReports,
      baseProps.reportSlot
    )
  })
})
