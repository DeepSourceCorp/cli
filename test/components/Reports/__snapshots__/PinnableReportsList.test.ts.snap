// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[[ PinnableReportsList ]] renders the list of pinnable reports as menu items 1`] = `"<span class=\\"z-menu relative\\"><div class=\\"leading-none\\"><zbutton-stub button-type=\\"ghost\\" color=\\"vanilla-400\\" icon=\\"settings-2\\" size=\\"small\\" class=\\"v-popper--has-tooltip\\"></zbutton-stub></div> <!----></span>"`;
